src/cmd-buildupload

```diff
@@ -164,8 +160,19 @@ def s3_check_exists(bucket, key):
 
 @retry(stop=retry_stop, retry=retry_boto_exception, retry_error_callback=retry_callback)
 def s3_copy(src, bucket, key, max_age, acl, extra_args={}, dry_run=False):
-    if key.endswith('.json') and 'ContentType' not in extra_args:
-        extra_args['ContentType'] = 'application/json'
+    extra_args = dict(extra_args)
+    if 'ContentType' not in extra_args:
+        if key.endswith('.json'):
+            extra_args['ContentType'] = 'application/json'
+        elif key.endswith('.tar'):
+            extra_args['ContentType'] = 'application/x-tar'
+        elif key.endswith('.xz'):
+            extra_args['ContentType'] = 'application/x-xz'
+        elif key.endswith('.gz'):
+            extra_args['ContentType'] = 'application/gzip'
```

Comment:
We could also just outsource all of this to `subprocess.check_output(['file', '-b', '--mime-type', key])` ?

Reply:
That was my initial thought, though there's been some hesitation around using [`file --mime-type`](https://github.com/coreos/coreos-installer/pull/36#issuecomment-494517948) in the past, so I stayed away from it. Also, it doesn't return the right type we want all the time. E.g. for gzip files, it returns `application/x-gzip` instead of `application/gzip`.